packages/runtime/src/helpers/utils.ts

```diff
@@ -74,7 +74,7 @@ export const generateNetlifyRoutes = ({
 }) => [...(withData ? toNetlifyRoute(dataRoute) : []), ...toNetlifyRoute(route)]
 
 export const routeToDataRoute = (route: string, buildId: string, locale?: string) =>
-  `/_next/data/${buildId}${locale ? `/${locale}` : ''}${route === '/' ? '/index' : route}.json`
+  `/_next/data/${buildId}${locale ? `/${locale}` : ''}${route === '/' ? (locale ? '' : '/index') : route}.json`
```

Comment:
yeah, I agree. i did try to refactor this but ended up in a knot because it really needs to broken up into 2 different functions - one for i18n and one for non-i18n - but it's difficult because of the various places it gets called. hopefully all this will be replaced soon with edge routing and/or improved CDN redirects